slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -94,6 +94,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
         else l
       }
     }
+
+    //Reference: https://github.com/slick/slick/issues/1419
+    override def createTableNamer(meta: MTable): TableNamer = new TableNamer(meta){
+      override def schema = meta.name.catalog
+      override def catalog = meta.name.schema 
```

Comment:
I doubt if this would make the right fix.

I have searched for information on this issue and this SO post comes to me: http://stackoverflow.com/questions/30567224/java-databasemetadata-getschemas-returns-empty-resultset-expected-non-empty-r
Here are some quotations:
> What MySQL calls schemas is more like catalogs defined in the SQL standard.
> The `getSchemas` method in Connector/J always returns an empty result set.
> The `getCatalogs` returns the result of `SHOW DATABASES`.

So it seems to me that it is not that the definitions of catalog and schema are reversed in MySQL JDBC driver, it is that there are only schemas in MySQL and they are called catalogs in the corresponding JDBC driver.

And if you look at #1419, it says that "the schema name is being set into the `catalog` field, and `None` is being set into the `schema` field." It does not necessarily mean that the catalog is being set into the `schema` field.

Perhaps more experiments/tests are required to better understand this issue?